Shoko.Server/AniDB_API/GroupStatusCollection.cs

```diff
@@ -14,30 +15,14 @@ public List<Raw_AniDB_GroupStatus> Groups
 
         public List<Raw_AniDB_GroupStatus> GetMissingEpisodes(int latestEpNumber)
         {
-            List<Raw_AniDB_GroupStatus> missingGroups = new List<Raw_AniDB_GroupStatus>();
-
-            foreach (Raw_AniDB_GroupStatus grp in groups)
-            {
-                if (grp.LastEpisodeNumber > latestEpNumber)
-                    missingGroups.Add(grp);
-            }
-
-            return missingGroups;
+	        return groups.Where(grp => grp.LastEpisodeNumber > latestEpNumber).ToList();
         }
 
         public int LatestEpisodeNumber
         {
             get
             {
-                int latest = 0;
-
-                foreach (Raw_AniDB_GroupStatus grp in groups)
-                {
-                    if (grp.LastEpisodeNumber > latest)
-                        latest = grp.LastEpisodeNumber;
-                }
-
-                return latest;
+                return groups.Select(grp => grp.LastEpisodeNumber).Concat(new[] {0}).Max();
```

Comment:
Old version returns 1+ items, i.e. groups that don't have episodes with max (id? number?), new version always returns exactly one result. Doesn't look right to me.